### DISCUSSION TITLE
Where to get FSharp.Compiler.Interactive.Settings.dll ?

### DISCUSSION THREAD

**Discussion by goswinr** [dotnet/fsharp#12599] [Q&A] [answered]
I am using fsi via FSharp.Compiler.Service 41.0.1 on net48 Framework.
How can I add the corresponding `FSharp.Compiler.Interactive.Settings.dll` to access the fsi object?
I read about [a workaround to copy it from a Visual Studio installation](https://github.com/dotnet/fsharp/issues/7950#issuecomment-565449011), but my intuition is that it should be a nuget package or part of the FCS package ( just like `FSharp.DependencyManager.Nuget.dll`).

Currently, just setting `useFsiAuxLib` to `true` in  
`let fsiConfig = FsiEvaluationSession.GetDefaultConfiguration(settings,useFsiAuxLib=true)` raises a `StopProcessingExn` because it can't find the Settings.dll in the same folder as FCS.

**Comment by vzarytovskii** [comment]
@KevinRansom @cartermp fyi

**Comment by goswinr** [comment, marked as answer]
Copying a `FSharp.Compiler.Interactive.Settings.dll` did not work for me.
The script has to reference the current `FSharp.Compiler.Service.dll`
and then get the `fsi` object at `FSharp.Compiler.Interactive.Shell.Settings.fsi`
see also [FSharpFar](https://github.com/nightroman/FarNet/blob/b89e39cad6e834aa296ee978ae1018ed3507d487/FSharpFar/src/About-FSharpFar.md#fsi-object)
